source/_posts/2019-10-14-hexo-4-released.md

```diff
@@ -0,0 +1,123 @@
+- fix: encode permalink by default [#3708]
+  * If you currently use `encodeURI(post.permalink)` (including `permalink` of page, tag & category variables), there are three options:
+    1. Use `encodeURI(decodeURI(post.permalink))` for backward-compatibility with hexo v3.9 (and older]
+    2. Use [`encodeURL()`](https://github.com/hexojs/hexo-util#encodeurlstr) function provided by hexo-util, it is backward-compatible
+    3. Drop `encodeURI()` function, this breaks backward-compatibility
+  * If you want to use the variables in decoded form,
+    - Use `decodeURI(post.permalink)`
+    - [`decodeURL()`](https://github.com/hexojs/hexo-util#decodeurlstr) of hexo-util can decode [punycoded](https://en.wikipedia.org/wiki/Punycode) domain.
+  * This change does not apply to `this.url` variable.
+- fix(paginator): add `escape` option for compatibility with hexo-util@1.3.0 [#3728]
+  * Theme devs, if you customize `prev_text`/`next_text` with html (e.g. to insert icons), you need to set [`escape: false`](https://github.com/theme-next/hexo-theme-next/blob/7d6272274afcc88edcefe3504e452a502c05329f/layout/_partials/pagination.swig#L8) in the parameter.
+
+## Feature
+
+- feat(filter): use existing excerpt if possible [#3612]
+  * Support `excerpt:` in [front-matter](https://hexo.io/docs/front-matter)
+- switch minimatch to micromatch [#3538]
+  * Refer to [micromatch docs](https://github.com/micromatch/micromatch#extended-globbing) for advanced globbing
+- feat: add option to disable meta generator tag [#3653]
+  * Add `meta_generator: false` to config to disable
+- feat(generator): allow limit parallel generation [#3665]
+  * If you experience "Out of memory" issue, try lowering the value in `hexo g --concurrency <number>`
+- Option to use date instead of file mtime for updated date [#3235]
+  * If you prefer not to use file modification time in the `post.updated` variable, set `use_date_for_updated: true` in config
+- feat(list_tags): add Schema "keywords" and Microdata "tag" [#3678]
+- feat(permalink_variable): add pretty_urls option to remove index.html from url [#3691]
+  * Use the following config to remove the trailing `index.html` from `permalink` [variables](https://hexo.io/docs/variables)
+
+  ``` yml
+  pretty_urls:
+    trailing_index: false
+  ```
+
+  * Does not apply to `this.url` variable (see [#3661]).
+  * Should be compatible with existing canonical-related plugins, recommend plugin and theme devs to test
+- feat: add `full_url_for` helper [#3701]
+  * use this helper `full_url_for(page.path)` instead of `config.url + page.path`
+- fix: external_link should use after_render [#3675]
+  * added `field` and `exclude` options, see [#3675] for guide.
+
+## Fix
+
+- fix: set english as default [#3654]
+- perf(meta_generator): drop cheerio [#3671]
+- perf(open_graph): drop cheerio and use regex [#3680]
+- perf(external_link): drop cheerio and use regex [#3685]
```

Comment:
IMHO these changes divide into another section is better. (e.g `Improvement` or `Performance` etc...).